opensearch-operator/api/v1/opensearchism_types.go

```diff
@@ -46,8 +46,8 @@ type OpenSearchISMPolicySpec struct {
 	Description       string             `json:"description"`
 	ErrorNotification *ErrorNotification `json:"errorNotification,omitempty"`
 	// Specify an ISM template pattern that matches the index to apply the policy.
-	ISMTemplate *ISMTemplate `json:"ismTemplate,omitempty"`
-	PolicyID    string       `json:"policyId,omitempty"`
+	ISMTemplate []ISMTemplate `json:"ismTemplate,omitempty"`
```

Comment:
Please do not change the CRD. Changing an existing field is an incompatible change that needs to be avoided at all costs and will certainly not be done for one minor thing.
Any needed translation  should be handled inside the operator.

Reply:
How do you propose to handle this in GO? If we need to be backwards compatible, we need to handle ISMTemplate both as an object or an array.
I can do some magic with `json.rawmessage` in the operator, but it seems very hacky.

Reply:
```suggestion
	ISMTemplate json.RawMessage `json:"ismTemplate,omitempty"`
```

Reply:
The issue is that the operator gateway returns []ISMTemplate

Reply:
Without having thought about it in detail: In the CRD keep it as an object, in the gateway change it to array and in the logic do the conversion (coming from the CRD make an array, coming from the gateway response only use the first element of the array). Not perfect but IMO better than making an incompatible change in the CRD.

Reply:
@swoehrl-mw I think i put too much thought into this. Changing the field type on the gateway struct alone seems sufficient